tests/proxy/pn_counter/pn_counter_test.go

```diff
@@ -0,0 +1,210 @@
+	if counterName != counter.PartitionKey() {
+		t.Errorf("PNCounter.PartitionKey failed")
+	}
+}
+
+func TestPNCounter_Destroy(t *testing.T) {
+	var delta int64 = 5
+	counter.AddAndGet(delta)
+	counter.Destroy()
+	counter, _ = client.GetPNCounter(counterName)
+	res, err := counter.Get()
+	AssertEqualf(t, err, res, int64(0), "PNCounter.Destroy failed")
+}
+
+func TestPNCounter_Get(t *testing.T) {
+	defer destroyAndCreate()
+	var delta int64 = 5
+	counter.AddAndGet(delta)
+	currentValue, err := counter.Get()
+	AssertEqualf(t, err, currentValue, delta, "PNCounter.Get failed")
+}
+
+func TestPNCounter_GetAndAdd(t *testing.T) {
+	defer destroyAndCreate()
+	var delta int64 = 5
+	previousValue, err := counter.GetAndAdd(delta)
+	AssertEqualf(t, err, previousValue, int64(0), "PNCounter.GetAndAdd failed")
+}
+
+func TestPNCounter_AddAndGet(t *testing.T) {
+	defer destroyAndCreate()
+	var delta int64 = 5
+	updatedValue, err := counter.AddAndGet(delta)
+	AssertEqualf(t, err, updatedValue, delta, "PNCounter.AddAndGet failed")
+}
+
+func TestPNCounter_GetAndSubtract(t *testing.T) {
+	defer destroyAndCreate()
+	var delta int64 = 5
+	previousValue, err := counter.GetAndSubtract(delta)
+	AssertEqualf(t, err, previousValue, int64(0), "PNCounter.GetAndSubtract failed")
+}
+
+func TestPNCounter_SubtractAndGet(t *testing.T) {
+	defer destroyAndCreate()
+	var delta int64 = 5
+	updatedValue, err := counter.SubtractAndGet(delta)
+	AssertEqualf(t, err, updatedValue, -delta, "PNCounter.SubtractAndGet failed")
+}
+
+func TestPNCounter_DecrementAndGet(t *testing.T) {
+	defer destroyAndCreate()
+	updatedValue, err := counter.DecrementAndGet()
+	AssertEqualf(t, err, updatedValue, int64(-1), "PNCounter.DecrementAndGet failed")
+}
+
+func TestPNCounter_IncrementAndGet(t *testing.T) {
+	defer destroyAndCreate()
+	updatedValue, err := counter.IncrementAndGet()
+	AssertEqualf(t, err, updatedValue, int64(1), "PNCounter.IncrementAndGet failed")
+}
+
+func TestPNCounter_GetAndDecrement(t *testing.T) {
+	defer destroyAndCreate()
+	previousValue, err := counter.GetAndDecrement()
+	AssertEqualf(t, err, previousValue, int64(0), "PNCounter.GetAndDecrement failed")
+}
+
+func TestPNCounter_GetAndIncrement(t *testing.T) {
+	defer destroyAndCreate()
+	previousValue, err := counter.GetAndIncrement()
+	AssertEqualf(t, err, previousValue, int64(0), "PNCounter.GetAndIncrement failed")
+}
+
+func TestPNCounter_ManyAdd(t *testing.T) {
+	defer destroyAndCreate()
+	var wg sync.WaitGroup
+	var delta = 1000
+	wg.Add(delta)
+	for i := 0; i < delta; i++ {
+		go func() {
+			counter.IncrementAndGet()
+			wg.Done()
+		}()
+	}
+	wg.Wait()
+	currentValue, err := counter.Get()
+	AssertEqualf(t, err, currentValue, int64(delta), "PNCounter has race condition")
+
+}
+
+func TestPNCounter_HazelcastNoDataMemberInClusterError(t *testing.T) {
+	client.Shutdown()
+	remoteController.ShutdownCluster(cluster.ID)
+	cluster, err = remoteController.CreateCluster("3.10", LiteMemberConfig)
```

Comment:
As far as I know, supplying a version number to `CreateCluster` does not have an effect in terms of server version number. It implies an effect that is actually not there. Moreover, these tests are going to run with later versions too. Maybe we should let it be `nil`.

Reply:
`string` can't be `nil` but I will make it `""`.